rb_ws/src/buggy/buggy/buggy_state_converter.py

```diff
@@ -0,0 +1,165 @@
+            return
+
+        # Publish the converted message to self/state
+        self.self_state_publisher.publish(converted_msg)
+
+    def other_raw_state_callback(self, msg):
+        """ Callback for processing other/raw_state messages and publishing to other/state """
+        # Convert the /SC message and publish to other/state
+        converted_msg = self.convert_SC_state(msg)
+        self.other_state_publisher.publish(converted_msg)
+
+    def convert_SC_state(self, msg):
+        """ Converts /SC namespace raw state to clean state units and structure """
+        converted_msg = Odometry()
+        converted_msg.header = msg.header
+
+        # ---- 1. Convert ECEF Position to UTM Coordinates ----
+        ecef_x = msg.pose.pose.position.x
+        ecef_y = msg.pose.pose.position.y
+        ecef_z = msg.pose.pose.position.z
+
+        # Convert ECEF to UTM
+        utm_x, utm_y, _ = self.ecef_to_utm_transformer.transform(ecef_x, ecef_y, ecef_z)
+        converted_msg.pose.pose.position.x = utm_x  # UTM Easting
+        converted_msg.pose.pose.position.y = utm_y  # UTM Northing
+        converted_msg.pose.pose.position.z = 0.0    # ignored
+
+        # ---- 2. Convert Quaternion to Heading (Radians) ----
+
+        qx, qy, qz, qw = msg.pose.pose.orientation.x, msg.pose.pose.orientation.y, msg.pose.pose.orientation.z, msg.pose.pose.orientation.w
+
+        # replaced with tf_transformations library function
+        # yaw = self.quaternion_to_yaw(qx, qy, qz, qw)
+        # converted_msg.pose.pose.orientation.x = yaw
+
+        # Use euler_from_quaternion to get roll, pitch, yaw
+        _, _, yaw = euler_from_quaternion([qx, qy, qz, qw])
+        yaw_aligned_east = yaw - np.pi / 2   # TODO: check if 0 is East
+        # Store the heading in the x component of the orientation
+        converted_msg.pose.pose.orientation.x = yaw_aligned_east
+
+        # ignored:
+        converted_msg.pose.pose.orientation.y = 0.0
+        converted_msg.pose.pose.orientation.z = 0.0
+        converted_msg.pose.pose.orientation.w = 0.0
+
+        # ---- 3. Copy Covariances (Unchanged) ----
+        # TODO: check whether covariances should be copied over or not sent at all
+        # converted_msg.pose.covariance = msg.pose.covariance
+        # converted_msg.twist.covariance = msg.twist.covariance
+
+        # ---- 4. Copy Linear Velocities ----
+        converted_msg.twist.twist.linear.x = msg.twist.twist.linear.x   # m/s in x-direction
+        converted_msg.twist.twist.linear.y = msg.twist.twist.linear.y   # m/s in x-direction
+        converted_msg.twist.twist.linear.z = msg.twist.twist.linear.z   # Keep original Z velocity (??)
+
+        # ---- 5. Copy Angular Velocities ----
+        converted_msg.twist.twist.angular.x = msg.twist.twist.angular.z   # rad/s for heading change rate (using yaw from twist.angular)
+        converted_msg.twist.twist.angular.y = 0.0   # ignored
```

Comment:
You don't need to ignore it, just copy it on over. more data, isn't a problem necessarily